solarforecastarbiter/io/fetch/nwp.py

```diff
@@ -0,0 +1,669 @@
+"""
+Fetch NWP files from NCEP Nomads for select variables. Should primarily be used
+as a CLI program.
+"""
+import asyncio
+import argparse
+from functools import partial
+import logging
+from pathlib import Path
+import re
+import shutil
+import signal
+import subprocess
+import sys
+import tempfile
+
+
+import aiohttp
+import pandas as pd
+import xarray as xr
+
+
+from solarforecastarbiter.io.fetch import (
+    handle_exception, basic_logging_config, make_session,
+    run_in_executor, start_cluster, abort_all_on_exception)
+
+
+logger = logging.getLogger(__name__)
+
+
+CHECK_URL = 'https://nomads.ncep.noaa.gov/pub/data/nccf/com/{}/prod'
+BASE_URL = 'https://nomads.ncep.noaa.gov/cgi-bin/'
+DOMAIN = {'subregion': '',
+          'leftlon': -126,
+          'rightlon': -66,
+          'toplat': 50,
+          'bottomlat': 24}
+
+
+def _gfs_valid_hr_gen(init_hr):
+    i = 0  # should we start at 1 and avoid the analysis?
+    while True:
```

Comment:
I have a hard time trusting `while True` blocks but I guess the code here is clear enough. I'd prefer something like `itertools.chain`

```python
>>> from itertools import chain
>>> c = chain(range(120), range(120, 240, 3), range(240, 385, 12))
>>> next(c)
0
>>> next(c)
1
```